src/main/java/net/blay09/mods/cookingforblockheads/container/comparator/ComparatorSoL.java

```diff
@@ -1,18 +1,44 @@
 package net.blay09.mods.cookingforblockheads.container.comparator;
 
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
 import java.util.Comparator;
+import java.util.function.BiFunction;
 
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 
 import squeek.spiceoflife.ModConfig;
 import squeek.spiceoflife.foodtracker.FoodHistory;
+import squeek.spiceoflife.foodtracker.foodqueue.FoodQueue;
 import squeek.spiceoflife.helpers.FoodHelper;
 
 public class ComparatorSoL implements Comparator<ItemStack> {
 
     private final ComparatorName fallback = new ComparatorName();
     private final EntityPlayer entityPlayer;
+    public static boolean spiceCompat = false;
+    private static MethodHandle getHistoryCompat;
+    private static BiFunction<ItemStack, FoodHistory, Boolean> getEverEaten;
+    static {
+        try {
+            getHistoryCompat = MethodHandles.lookup()
+                    .findVirtual(FoodHistory.class, "getHistory", MethodType.methodType(FoodQueue.class));
+            getEverEaten = (itemstack, history) -> {
+                try {
+                    return ((FoodQueue) getHistoryCompat.invokeExact(history)).stream()
+                            .anyMatch(foodEaten -> foodEaten.itemStack.isItemEqual(itemstack));
+                } catch (Throwable ignored) {
```

Comment:
Please don't catch `Throwable`s, only `Exception`s.

Reply:
How else should I handle it? invokeExact throws specifically Throwable, so I can't just catch Exception